compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1440,6 +1440,225 @@ LogicalResult ArgCompareOp::generateScalarImplementation(OpBuilder &b,
+    } else {
+      // For non-reduction dimensions, use the same offsets/sizes as input.
+      initOffsets.push_back(offsets[i]);
+      initSizes.push_back(sizes[i]);
+    }
+    initStrides.push_back(b.getIndexAttr(1));
+  }
+
+  const ShapedType &initValType = getOutputValueType();
+  const ShapedType &initIdxType = getOutputIndexType();
+  SmallVector<int64_t> resultValShape, resultIdxShape;
+  for (int64_t i = 0; i < rank; ++i) {
+    if (i == reductionDim)
+      continue;
+
+    if (auto sizeAttr = dyn_cast<Attribute>(initSizes[i])) {
+      int64_t size = (cast<IntegerAttr>(sizeAttr)).getInt();
+      resultValShape.push_back(size);
+      resultIdxShape.push_back(size);
+      continue;
+    }
+
+    resultValShape.push_back(ShapedType::kDynamic);
+    resultIdxShape.push_back(ShapedType::kDynamic);
+  }
+
+  RankedTensorType sliceValResultType =
+      RankedTensorType::get(resultValShape, initValType.getElementType(),
+                            cast<RankedTensorType>(initValType).getEncoding());
+  RankedTensorType sliceIdxResultType =
+      RankedTensorType::get(resultIdxShape, initIdxType.getElementType(),
+                            cast<RankedTensorType>(initIdxType).getEncoding());
+
+  auto initValSlice = tensor::ExtractSliceOp::create(
+      b, loc, sliceValResultType, init[0], initOffsets, initSizes, initStrides);
+  tiledOperands.push_back(initValSlice.getResult());
+  slices.push_back(initValSlice);
+
+  auto initIdxSlice = tensor::ExtractSliceOp::create(
+      b, loc, sliceIdxResultType, init[1], initOffsets, initSizes, initStrides);
+  tiledOperands.push_back(initIdxSlice.getResult());
+  slices.push_back(initIdxSlice);
+
+  // Create index_base for this chunk.
+  Value tileIndex =
+      getValueOrCreateConstantIndexOp(b, loc, splitReductionIvs[0]);
+  Value tileSize = getValueOrCreateConstantIndexOp(b, loc, sizes[reductionDim]);
+  Value tileStartIndex = b.create<arith::MulIOp>(loc, tileIndex, tileSize);
+
+  Value newIndexBase;
+  if (Value globalIndexBase = getIndexBase()) {
+    // Add chunk start to existing index_base.
+    newIndexBase =
+        b.create<arith::AddIOp>(loc, globalIndexBase, tileStartIndex);
+  } else {
+    // Use chunk start as index_base.
+    newIndexBase = tileStartIndex;
+  }
+
+  SmallVector<Type> resultTypes;
+  if (hasPureTensorSemantics()) {
+    resultTypes.push_back(initValSlice->getResult(0).getType());
+    resultTypes.push_back(initIdxSlice->getResult(0).getType());
+  }
+
+  // Create the tiled operation with adjusted index_base.
+  SmallVector<Value> operands = std::move(tiledOperands);
+  operands.push_back(newIndexBase);
+  Operation *tiledArgmaxOp = b.create<ArgCompareOp>(
+      loc, resultTypes,
+      operands[0],                          // input slice.
+      ValueRange{operands[1], operands[2]}, // init slices.
+      operands[3],                          // index_base.
+      reductionDim                          // dimension.
```

Comment:
Use the `/*argName=*/ paramName` format. Clang tidy and other linters can check that the names are up to date